utbot-intellij/src/main/kotlin/org/utbot/intellij/plugin/generator/UtTestsDialogProcessor.kt

```diff
@@ -181,243 +183,248 @@ object UtTestsDialogProcessor {
             .map { it.containingFile.virtualFile }
             .toTypedArray()
 
-        val promise = compile(project, filesToCompile, springConfigClass)
-        promise.onSuccess {
-            if (it.hasErrors() || it.isAborted)
-                return@onSuccess
+        model.preCompilePromises
+            .all()
+            .thenRun {
+                val compilationPromise = compile(project, filesToCompile, springConfigClass)
```

Comment:
Using fluent promise API can help avoid additional layer of nesting here
```suggestion
        val promise = model.preCompilePromises.all().thenAsync {
            compile(project, filesToCompile, springConfigClass)
        }
```